### ISSUE TITLE
Separate positions for the dots with the indicator for the current application

### ISSUE THREAD

**Issue by swarnendubiswas** [jderose9/dash-to-panel#12]
This is a useful extension. I think it can be better if it is possible to separate the position for the dots and the indicator for the current application. In the attached snapshot, I have two Skype windows open, but it cannot be inferred from the panel.

![screenshot from 2017-01-08 23-26-58](https://cloud.githubusercontent.com/assets/7343807/21792463/9f219860-d6af-11e6-992c-fb9f31ea2225.png)

**Comment by jderose9** [comment]
Yes, I agree. I'm having trouble coming up with a decent solution that doesn't look too cluttered, though. Any ideas?

I think it is best to not depart too drastically from the stock Gnome design as many themes adjust the style of the running indicator and if we don't re-use those styles things may look out of place.

**Comment by swinkl3** [comment]
I would also like to see this. Something like the below would be my suggestion. This combines both the number of dots from unfocused windows as well as the line from the focused window. This takes existing resources and still allows for styling based on theme.

![dash-to-panel](https://cloud.githubusercontent.com/assets/19921181/21793948/07c5930e-d74a-11e6-90fc-70835420b10e.png)

**Comment by fastrizwaan** [comment]
![test](https://cloud.githubusercontent.com/assets/1161682/21800994/e195fd58-d745-11e6-9174-477b1832623f.png)

The bar split into 2 or 3 (if three or more instances running)

**Comment by zagortenay333** [comment]
The correct way to solve this is to add the corresponding dash-to-dock style classes and containers to this extension; that way the exact styling of the dash-to-dock extension would be applied here as well.

Themes that style the dash to dock would support this as well automatically.

**Comment by jderose9** [comment]
@zagortenay333 I don't think this is possible to pull everything in (I just gave more detail in #15). I also (personally) prefer the styles pulled in from zorin as a default rather than those in dash-to-dock. While it's probably possible to remove the extra margin around the container in dash-to-dock, I've mostly seen themes use both a top and bottom indicator to address this problem, which seems very cluttered to me.

It would be possible to apply additional classes the way dash-to-dock has to indicate the running count (running<X>).

**Comment by jderose9** [comment]
Is it actually useful to know exactly how many windows are open or just that there are multiple?

KDE Plasma uses an overlay indicator like this:
![screenshot_20170110_090926](https://cloud.githubusercontent.com/assets/2118702/21809243/d3b5b3a8-d714-11e6-92a5-ed95daaed906.png)

**Comment by jderose9** [comment]
For reference, here's windows.. They add shade a segment of the bar to indicate multiple open windows.
![untitled](https://cloud.githubusercontent.com/assets/2118702/21818880/7bff043e-d737-11e6-842a-9bfe25c05f81.png)

**Comment by swarnendubiswas** [comment]
This is personal taste, but I prefer numbers (if < 10). They are more legible than discerning dots and shades. :)

**Comment by jderose9** [comment]
Here's some more feedback about this from @gnunn1 in #17:

> I have a hard time seeing the dots that are used for background applications. Would it be possible to make them a bit wider then they are now? Admittedly part of this is because I use a theme with a darker color for selection color, as a result it doesn't standout well against the black background of the panel.

**Comment by brandleesee** [comment]
I suggest that the enlarged and widened dots be clickable in such a way that on click they raise the corresponding window (which can be 'calculated' by having the leftmost dot as the first window opened etc). This would go hand in hand with window previews. If this is already coded, than please ignore my comment.

**Comment by brandleesee** [comment]
![screenshot from 2017-01-18 09-16-50](https://cloud.githubusercontent.com/assets/12674437/22056130/e1cc8ff2-dd5e-11e6-9864-4c47f6d0c5c4.png)

To further illustrate the point of this thread:
There are two instances of Google Chrome opened.
And there are also two instances of Nautilus opened.
However, the two minimised chrome windows are dotted but the two visible nautilus windows only show one think unbroken line (not two).

Also, could the _size of the minimised dots become a setting for increasing their size_?
They are too small when compared to the icons' size and thick line.
Thanks.

**Comment by zagortenay333** [comment]
Here is how I did it in my themes, both in the _dash-to-dock_ and _dash-to-panel_ extensions:

![selection_004](https://cloud.githubusercontent.com/assets/11144630/22065679/9c91c706-dd89-11e6-874c-d3cbfc5d2797.png)

The point being that you shouldn't use the bar to indicate what app is active. Use a background to indicate which app is active, while the bars indicate what apps are running and can the be split in some way to indicate how many apps are running.

It looks more consistent to me then the mixture of bars and dots.

Also an _optional_ number to indicate how many apps are running would be nice too.

**Comment by brandleesee** [comment]
@jderose9 
> a decent solution that doesn't look too cluttered
I think @zagortenay333's method is the soundest visually appealing display brought forward here. I find it exquisitely stylish.

@zagortenay333 Would you consider sharing your code through a pull-request? (sorry for asking)

Why the fuss, you ask?
Well, because I was using Taskbar but it had a huge number of settings which were too much to plow through. I used two other extensions which did not get the job done (very well). A couple of months after removing it, dash-to-panel enters the scene.

**Comment by zagortenay333** [comment]
This is actually implemented in my ciliora themes:
https://github.com/zagortenay333/ciliora-tertia-shell
https://github.com/zagortenay333/ciliora-secunda-shell

That's why I've been harassing OP with theme support requests lately. :stuck_out_tongue_closed_eyes: 
Any theme makers can do it too.

**Comment by Pointedstick** [comment]
@zagortenay333 that is awesome-looking and I too would love to see that made the default. Lightening the background to show the frontmost app is perfect.

**Comment by zagortenay333** [comment]
@jderose9 Yeah, it looks exactly how I wanted it.

The running indicators can't be flipped because I'm not using the `.app-well-app-running-dot` actor:
https://github.com/zagortenay333/ciliora-tertia-shell/blob/1194c2a1b95b39a141901c02d5584eb413093a97/Ciliora-Tertia/gnome-shell/scss/extensions/_dash-to-panel.scss

I guess it could be solved by adding a class to the `.app-well-app` container.

**Comment by brandleesee** [comment]
@jderose9 
> current behavior is to use the height of the running indicator line as the circle radius.... that's not very visible for themes with a thinner indicator

How can this be remedied: theme side or extension side?

**Comment by jderose9** [comment]
@brandleesee Right now, you can fix this in the theme by setting the height property in the "app-well-app-running-dot" class in the themes gnome-shell.css.

Ultimately, my goal is to provide defaults out of the box that work with most popular themes, then let the theme override that if they want, and finally let the user tweak key settings that may not be one-size-fits-all (like indicator size). I'm not really sure if there's a bunch of users that are happy with the current appearance so I don't know whether it's worth trying to continue to support that too. Just need to get some time to work on it.

**Comment by jderose9** [comment]
@devfull said in #52:

> I would like to request an additional focus indication for open/active/inactive windows. The bar and dots used now are imo not clear enough, especially the dots (where I have to look closer at the panel to see them).
So I think, a square around the icons is comfortable for the eyes and looks good (my opinion). Am not requesting to replace the one used now, but rather an option to choose which one to use, so that people who like 'the bar/dots' can keep using it.
See screenshot for how it could look like ;)
![screenshot_20170128_145224](https://cloud.githubusercontent.com/assets/20864876/22397079/94aa195c-e569-11e6-888e-4487bb96af1f.png)

**Comment by Pointedstick** [comment]
I prefer the light grey background idea to the green border idea.

**Comment by jderose9** [comment]
Here's where I'm at with this. It's a little more subtle than a lot of the suggestions posted here, but I wanted to avoid anything that might clash with the existing popular themes. (If it's not clear, there's two terminals open and one file manager)

Arc
![screenshot from 2017-02-15 23-17-45](https://cloud.githubusercontent.com/assets/2118702/23007316/e9519960-f3d5-11e6-90bb-b04b6cfd54b6.png)
![screenshot from 2017-02-15 23-17-25](https://cloud.githubusercontent.com/assets/2118702/23007261/76776956-f3d5-11e6-8065-a3390570eb79.png)

Flat-Plat
![screenshot from 2017-02-15 23-29-46](https://cloud.githubusercontent.com/assets/2118702/23007451/c459241a-f3d6-11e6-8cdb-5a2cd0464271.png)
![screenshot from 2017-02-15 23-26-51](https://cloud.githubusercontent.com/assets/2118702/23007435/b38b1580-f3d6-11e6-94a4-c57cf8a347e3.png)

The main downside here is that you don't know how many windows are open - just that there is more than one.

In terms of configuration settings I was thinking:
- Configure color of the line
- Height of the line
- Disable the stacked appearance when multiple exist
- Disable the background highlight
- Switch the unfocused application indicator back to the current dots, or dashes instead.

So basically it would be possible to get back to the existing behavior if desired.

This wouldn't preclude having some sort of configurable badge icon that showed the number of open windows overlayed in the corner of the icon at some point later.

Let me know what you think and whether this addresses the existing problems. Thanks!

**Comment by gnunn1** [comment]
@jderose9 While it's an improvement, I much prefer the example that @zagortenay333 showed from his theme. If that's feasible it would be my preference because it's clear at a glance how many windows are open.

**Comment by brandleesee** [comment]
I second @gnunn1

**Comment by jderose9** [comment]
Ok, I've made some changes to the way that the indicators are being handled and I was able to incorporate both the new style I had proposed, @zagortenay333's style from his Ciliora theme, and a number of other options.

Here's the settings menu options:
![indicator_common](https://cloud.githubusercontent.com/assets/2118702/23236655/0985047c-f928-11e6-8072-5a17288ba3a2.png)
![indicator_detail](https://cloud.githubusercontent.com/assets/2118702/23236662/10110ea8-f928-11e6-91d3-9d594a4b1395.png)

Here's some different looks I achieved with the new settings:
![ciliora](https://cloud.githubusercontent.com/assets/2118702/23236677/1a25120e-f928-11e6-9d61-73339ad9ccfc.png)
![metro](https://cloud.githubusercontent.com/assets/2118702/23236680/1d899d3e-f928-11e6-9968-604f9a6826f6.png)
![dashes](https://cloud.githubusercontent.com/assets/2118702/23236685/2008eb00-f928-11e6-8529-343d11eb0d6c.png)
![squaressegmented](https://cloud.githubusercontent.com/assets/2118702/23236689/25a6f46c-f928-11e6-9844-ba1a4f9a3201.png)
![dotssolid](https://cloud.githubusercontent.com/assets/2118702/23236694/2bd2bf24-f928-11e6-83e9-1136d2ed9b07.png)

This stuff is in `master` so check it out and let me know what you think.

**Comment by gnunn1** [comment]
That looks great, thanks!

**Comment by franglais125** [comment]
Congrats for such a nice implementation, both for customization and looks!

**Comment by jderose9** [comment]
Should I close this or is something still missing? Does anyone still think badge/overlay style indicator with an actual number would be a useful feature?

**Comment by swarnendubiswas** [comment]
It is no longer a feature, but another customizability option. Low priority now it seems. Good job by the way!

**Comment by amankhoza** [comment]
I just updated, and find the new options amazing, highly configurable to keep everyone happy, great work :1st_place_medal:

**Comment by jderose9** [comment]
Thanks for the feedback, everyone!
